pkg/ccl/streamingccl/streamingest/stream_ingestion_frontier_processor.go

```diff
@@ -467,7 +467,11 @@ func (sf *streamIngestionFrontier) maybeUpdatePartitionProgress() error {
 	sf.metrics.JobProgressUpdates.Inc(1)
 	sf.persistedHighWater = f.Frontier()
 	sf.metrics.FrontierCheckpointSpanCount.Update(int64(len(frontierResolvedSpans)))
-	sf.metrics.FrontierLagSeconds.Update(timeutil.Since(sf.persistedHighWater.GoTime()).Seconds())
+	if sf.persistedHighWater != sf.highWaterAtStart {
```

Comment:
Out of curiosity, any reason not to compare sf.persistedHighWater to 0 instead?  Just so we don't have to think about what highWaterAtStart actually means in this context?

Reply:
Ah, good q. Suppose the stream ingestion job is paused after making some progress. Then during the initial scan on resume, I _think_ we want Replication Lag to be 0 and not balloon to some ridiculous number. If we just compared `persistedHighWater==0`, then this metric would blow up on resume. If you agree with my reasoning, I'll add a note to explain why we use `highWaterAtStart`.

Reply:
I'm not sure I agree.  Once we have a non-zero watermark, then the lag blowing up on resume is actually correct. That is how far behind you are, anything else is kinda a lie.

Reply:
Sounds good, I'll use persistedHighWater then.

In this case, we ought to also emit a metric with the latest status of the replication job (or any job in general). That way, if a customer wants to alert on high replication lag, they can filter the alert during an initial scan. I can write up an issue, as this seems like lower priority UX work.